payment_api/views.py

```diff
@@ -26,6 +30,10 @@ class PaymentNotificationAPIView(APIView):
     """
     authentication_classes = (OAuth2Authentication, SessionAuthentication)
 
+    @method_decorator(transaction.non_atomic_requests) # Used to add user into cohort
```

Comment:
Can you explain why do we need to do that?
You should probably only wrap the call to register the cohort and not the whole request which needs to be atomic.